x-pack/legacy/plugins/task_manager/task.ts

```diff
@@ -235,6 +235,11 @@ export interface TaskInstance {
  */
 export type TaskInstanceWithId = Require<TaskInstance, 'id'>;
 
+/**
+ * The Scheduling fields of a task instance that has an id.
+ */
+export type TaskInstanceScheduling = Pick<TaskInstanceWithId, 'id' | 'runAt' | 'interval'>;
```

Comment:
The name `TaskInstanceScheduling` isn't obvious at first about what it represents (maybe `RescheduleOpts`?). We could move this within the `task_store.ts` file and follow the flow done for `OwnershipClaimingOpts` https://github.com/elastic/kibana/blob/bf247018897c468ae2596f329b1fa325dc0cb523/x-pack/legacy/plugins/task_manager/task_store.ts#L59-L62.